src/gui/advancedsettings.cpp

```diff
@@ -436,6 +439,16 @@ void AdvancedSettings::loadAdvancedSettings()
     addRow(OS_MEMORY_PRIORITY, (tr("Process memory priority (Windows >= 8 only)")
         + ' ' + makeLink("https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/ns-processthreadsapi-memory_priority_information", "(?)"))
         , &m_comboBoxOSMemoryPriority);
+
+    m_spinBoxMemoryWorkingSetLimit.setRange(0, 100);
+    m_spinBoxMemoryWorkingSetLimit.setSingleStep(5);
+    m_spinBoxMemoryWorkingSetLimit.setSuffix(u"%"_qs);
+    m_spinBoxMemoryWorkingSetLimit.setSpecialValueText(tr("Auto"));
+    m_spinBoxMemoryWorkingSetLimit.setValue(static_cast<Application *>(QCoreApplication::instance())->memoryWorkingSetLimit());
+
+    addRow(MEMORY_WORKING_SET_LIMIT, (tr("Process memory working set limit")
+        + ' ' + makeLink("https://docs.microsoft.com/en-us/windows/win32/memory/working-set", "(?)"))
```

Comment:
I have intention to make it available on linux so I would suggest using https://en.wikipedia.org/wiki/Working_set
And please make adjustments for the option to be platform agnostic.

Reply:
> And please make adjustments for the option to be platform agnostic.

What adjustments? Allow to set it in settings UI on any platform? But currently it is implemented only for Windows. If someone implements it on some other platform, they can easily adjust the conditionals to make it available.

Reply:
OK. Another thing, are you able to add the webapi (or even webui) parts? AFAIK windows can build with nox/nogui option too.

Reply:
> OK. Another thing, are you able to add the webapi (or even webui) parts? AFAIK windows can build with nox/nogui option too.

I will do this as soon as the main part is finished and approved.

Reply:
> > OK. Another thing, are you able to add the webapi (or even webui) parts? AFAIK windows can build with nox/nogui option too.
> 
> I will do this as soon as the main part is finished and approved.

Now I see it's impossible since it is still OS dependent and we discussed earlier that the API should be unified.
In case you forgot, let me remind you that in the case of such platform-dependent things, we have two main ways: 
1. do not provide them through the API, 
3. Use some stubs in the API.

We can still, of course, have some kind of modifier in the API version, but it looks too troublesome.
Also, we may not hide the interface for platform-dependent functions, but simply do an implementation that does nothing where it is not available. For example, in the case of this option, we could just allow it to be get/set, but mark it as unsupported on such and such platforms.

Reply:
>Also, we may not hide the interface for platform-dependent functions, but simply do an implementation that does nothing where it is not available. 

This.

IMO you can just merge this PR and I'll add the linux and webapi parts.

Reply:
>IMO you can just merge this PR and I'll add the linux and webapi parts.

I have bad news. The relevant system API (`RLIMIT_RSS`) on linux has no effect on modern systems (only affects 2.4.x kernel versions) and there are no replacements. I reckon linux has no need of such tuning knob...
The same API still might benefit BSD, macOS but I don't use them and the user base on them are relatively small...
So for those users, maybe I'll just publish my changes in a new PR and close it immediately (since the effectiveness is dubious)?

Reply:
> So for those users, maybe I'll just publish my changes in a new PR and close it immediately (since the effectiveness is dubious)?

I'm not sure I understand exactly its meaning...

Reply:
>I'm not sure I understand exactly its meaning...

This is to leave a mark in the sea of closed PR so that users finding it would able to find something...

Reply:
>This is to leave a mark in the sea of closed PR so that users finding it would able to find something...

Ok.

Reply:
> > This is to leave a mark in the sea of closed PR so that users finding it would able to find something...
> 
> Ok.

#16576.